tests/providers/amazon/aws/executors/ecs/test_ecs_executor.py

```diff
@@ -598,6 +598,131 @@ def test_attempt_task_runs_attempts_when_some_tasks_fal(self, _, mock_executor,
             == caplog.messages[0]
         )
 
+    @mock.patch.object(ecs_executor, "calculate_next_attempt_delay", return_value=dt.timedelta(seconds=0))
+    def test_task_retry_on_api_failure_all_tasks_fail(self, _, mock_executor, caplog):
```

Comment:
Maybe mock the number of attempts to 2 instead of 3. That would simplify this test, but I do like how explicit it is!